daisy_workflows/export/export_disk_ext.sh

```diff
@@ -14,31 +14,56 @@
 # limitations under the License.
 set -x
 
+DISK_RESIZING_MON=disk_resizing_mon.sh
+
 URL="http://metadata/computeMetadata/v1/instance/attributes"
 GS_PATH=$(curl -f -H Metadata-Flavor:Google ${URL}/gcs-path)
 FORMAT=$(curl -f -H Metadata-Flavor:Google ${URL}/format)
 
 # Strip gs://
-LOCAL_PATH=${GS_PATH##*//}
+IMAGE_OUTPUT_PATH=${GS_PATH##*//}
 # Create dir for output
-OUTS_PATH=${LOCAL_PATH%/*}
+OUTS_PATH=${IMAGE_OUTPUT_PATH%/*}
 mkdir -p "/gs/${OUTS_PATH}"
 
-# Disk image size info.
+# Prepare disk size info.
+# 1. Disk image size info.
 SIZE_BYTES=$(qemu-img info --output "json" /dev/sdb | grep -m1 "virtual-size" | grep -o '[0-9]\+')
- # Round up to the next GB.
-SIZE_GB=$(awk "BEGIN {print int((${SIZE_BYTES}/1000000000)+ 1)}")
+# 2. Round up to the next GB.
+SIZE_OUTPUT_GB=$(awk "BEGIN {print int((${SIZE_BYTES}/1000000000) + 1)}")
+# 3. Add 5GB of additional space to max size to prevent the corner case that output
+# file is slightly larger than source disk.
+MAX_BUFFER_DISK_SIZE_GB=$(awk "BEGIN {print int(${SIZE_OUTPUT_GB} + 5)}")
+
+# Prepare buffer disk
+echo "GCEExport: Initializing buffer disk for qemu-img output..."
+sudo arted -a optimal /dev/sdc mklabel gpt
```

Comment:
parted

Reply:
there's no error checking on these commands..

Reply:
why create a label if you're putting a filesystem directly on the root disk btw?

Reply:
no need for a label, removed.
I didn't add error check for local operations since I assumed they will success (they have lower chance to fail, comparing to network calls), and if not, the following steps will fail due to file system problem.
Do you think it's valuable to add error check? I feel that made the whole process too complex.

Reply:
well what is the expected outcome if either mkfs or mklabel fail? what will log output?

Reply:
if mkfs fail, then mount will also fail. And the whole workflow ultimately failed on qemu outputing step. We can't get an accurate error msg in that case. I was trying to balance the probability of failure v.s. code complexity.

Shell scripts in this code base always contain a serial of linux commands. Per my observation, local command usually doesn't have an error handling logic. I think it's because the probability is low. An example: mounting a pd ssd doesn't have error handling, while mounting a GCS path via gcsfuse has error handling.

So I didn't add error handling for mkfs to keep the code simple and clear. But if mkfs actually has a reasonably-high probability to fail, then I totally agree to add error handling. How do you think?

Reply:
i think the logic for where to add error handling is related partly to failure risk, i.e. 'ls' command is unlikely to fail for files you have created. It is also related to logic of the program. In this case i'm not sure what the result will be, so I suggest we either clarify that we have confidence of the failure case (what will the failure mode be, how will it be presented) or add add'l checks.

Reply:
Sure! let me add clarification here